### ISSUE TITLE
Uploads Don't Follow S3 Configuration

### ISSUE THREAD

**Issue by Dan6erbond** [outline/outline#4102]
Hello, I have configured Outline as per the Docker configuration documentation, and want to connect it with a DigitalOcean Spaces Bucket for S3 uploads. 

Uploading a profile image and team logo work perfectly, but if I just insert an image into Outline it seems to use some hardcoded values to Amazon S3, because the image link includes `https://s3.fra1.amazonaws.com/outline-space/uploads/`.

My environment configuration for Outline looks like this:

![image](https://user-images.githubusercontent.com/32235074/189493076-47135c31-9602-4661-8737-55c0cd43573f.png)

What I found is that some of the images are uploaded to my DigitalOcean Spaces Bucket, but it seems Outline cannot load them correctly, so it removes the reference from the document. Might just be a simple error where somewhere the Amazon URLs are still hardcoded.

![image](https://user-images.githubusercontent.com/32235074/189493425-62bcfd19-261b-4719-aa1c-0570ebd6cd07.png)

**To Reproduce**
1. Go to any page
2. Paste an image, or drag-and-drop
3. See error

**Expected behavior**

The image should be uploaded and visible to others. It ends up loading endlessly.

**Screenshots**
If applicable, add screenshots or videos to help explain your problem.

![image](https://user-images.githubusercontent.com/32235074/189493134-7f87a3c3-b97a-41df-bd06-be33dd37097e.png)

**Outline (please complete the following information):**
- Install: self hosted
- Version: v0.66.0

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Brave - PWA

**Comment by Dan6erbond** [comment]
I managed to find the issue: Turns out Outline wanted the AWS S3 **Endpoint** URL, and not the _Bucket URL_, because the initialization of the S3 client [here](https://github.com/outline/outline/blob/0c51bfb8998a08e5565a9e9cd5c4f19a3fcd0df6/server/utils/s3.ts#L29) seems to set the endpoint to `undefined` if it contains the bucket name.

Since profile image uploading and team logo worked, I find this behavior along with he environment variable naming confusing. The documentation at least in the [guide for configuring AWS S3](https://wiki.generaloutline.com/share/125de1cc-9ff6-424b-8415-0d58c809a40f) should be clearer about this, and it might still be considered a bug that the bucket URL from services like DO isn't supported, but the endpoint URL with the path-style URL does work.
